CONTRIBUTING.md

```diff
@@ -36,21 +34,43 @@ To send us a pull request, please:
 5. Send us a pull request, answering any default questions in the pull request interface.
 6. Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
 
-GitHub provides additional document on [forking a repository](https://help.github.com/articles/fork-a-repo/) and
-[creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
+GitHub provides additional document on [forking a repository](https://help.github.com/articles/fork-a-repo/) and [creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
+
+
+## Coding conventions and Style guides
+
+In addition to pursuing code correctness, please also keep the following standards, guides, norms, and conventions in mind when drafting contributions.
+
+### Enforced via CI
+
+* [Staticcheck](https://staticcheck.io/)
```

Comment:
By default golangci staticcheck seems only check SAxxx checks:

https://golangci-lint.run/usage/linters/#staticcheck

We should mention this or use the golangci link above.

Reply:
maybe mention `make check` will help you ensure your changes pass these checks including linting, DCO, ltag etc.

Reply:
Source for only checking SAxxx checks? Link and https://staticcheck.io/docs/configuration/options/#checks suggest to me that it's all checks minus six specific ST checks. Also, we seem to actually want those excluded checks, so we should probably set this, but that's another issue.

Reply:
I got this from golangci's staticcheck config: https://golangci-lint.run/usage/linters/#staticcheck

```
    # SAxxxx checks in https://staticcheck.io/docs/configuration/options/#checks
    # Default: ["*"]
    checks: ["all"]
```

The comment mentioned `SAxxxx` checks, but the default value is `all` ... not sure which one is more accurate.

Also, for this staticcheck error, curious how you got this error? (e.g. via golangci run/make check, or others). Since it was not discovered by golangci in previous PRs. https://github.com/awslabs/soci-snapshotter/pull/442#discussion_r1113314577

Reply:
I think my local staticcheck might be running `["*"]` by default. I agree that the docs aren't very consistent, but you're right that my previous experience indicates that it must not be running every check since that thing wasn't caught. Also, other docs say that the statichcheck default is `["all", "-ST1..."]` which means that "all" would have to include the ST checks to make that syntax valid.

I'll add some notes to the doc and hopefully we won't get stuck here on diving into that particular bit.